src/intervals/flavors.jl

```diff
@@ -0,0 +1,80 @@
+"""
+    Flavor{F}
+
+Super type of all interval flavors.
+
+A flavor defines (following the IEEE Std 1788-2015) how an interval behave
+in edge cases. This mostly makes a difference when dealing with
+infinity and division by zero.
+
+Currently only Flavor{:set_based} is supported.
+
+- `:set_based` (default) : Elements of an interval are real number.
+    In particular, infinity is never part of an interval and is only used as a
+    shorthand.
+    For example, the interval `(2..Inf)` contain all real number greater than 2.
+    In particular, this means that `(Inf..Inf)` is an empty interval, and division
+    by a thin zero return the empty interval.
+    The edge cases are
+        - `x/(0..0) ≛ ∅`
+        - `(0..0)/(0..0) ≛ ∅`
+        - `(0..0)*(-Inf..Inf) ≛ 0`
+        - `Inf ∈ (0..Inf) == false`
+    This flavor is described and required in part 2 of the IEEE Std 1799-2015.
+- `:cset` (not implemented) : Elements of an interval are either real numbers
+    or `±Inf`, applying standard rule for arithmetic with infinity.
+    The edge cases are
+        - `x/(0..0) ≛ (-Inf..Inf)`
+        - `(0..0)/(0..0) ≛ (-Inf..Inf)`
+        - `(0..0)*(-Inf..Inf) ≛ (-Inf..Inf)`
+        - `Inf ∈ (0..Inf) == true`
+"""
```

Comment:
Naive question (looking into the future): how is the user supposed to change flavors?

Reply:
By redefining `IntervalArithmetic.current_flavor()`.